### DISCUSSION TITLE
Use the GPU power observer for a Fortran-based program

### DISCUSSION THREAD

**Discussion by sjsprecious** [KernelTuner/kernel_tuner#229] [Q&A] [answered]
I attended SC23 and found this powerful tool for performance tuning on the GPU.

I have a Fortran-based program (using OpenACC for GPU offloading) and wonder whether I am able to use this tool to measure the GPU power consumption (i.e., I just need the GPU power (unit: W) or energy (unit: J) data, no need for tuning other parameters).

**Comment by isazi** [comment]
Hi @sjsprecious while you could use Kernel Tuner to do that if you wanted, if you are not going to tune anything and you just want to measure the energy consumption this may be overkill. Are you interested in the energy consumption of each kernel, or the full program consumption?

**Comment by sjsprecious** [comment]
Hi @isazi, thanks for your reply. I am more interested in the energy consumption of a full program that runs on the GPU. Do you know how I can measure it? I found some C++ examples online but not for Fortran so far.

**Comment by isazi** [comment]
Maybe something like EAR [1]? It is a daemon running on the system, so it works with any executable and provides you with energy measurements.

[1] https://www.bsc.es/research-and-development/software-and-apps/software-list/ear-energy-management-framework-hpc

**Comment by benvanwerkhoven** [comment, marked as answer]
I think PMT (https://git.astron.nl/RD/pmt/) could work here. It's quite easy to interface C code from Fortran. Or you could directly use NVML of course.

**Comment by sjsprecious** [comment]
Thanks @isazi and @benvanwerkhoven for your suggestions. That is very helpful!
